coordinator/txmanager.go

```diff
@@ -323,16 +329,21 @@ func (t *TxManager) checkEthTransactionReceipt(ctx context.Context, batchInfo *B
 func (t *TxManager) handleReceipt(ctx context.Context, batchInfo *BatchInfo) (*int64, error) {
 	receipt := batchInfo.Receipt
 	if receipt != nil {
-		if batchInfo.EthTx.Nonce()+1 > t.accNonce {
-			t.accNonce = batchInfo.EthTx.Nonce() + 1
+		// taking last eth tx nonce as it probably have the highest nonce
+		lastEthTx := batchInfo.EthTxs[len(batchInfo.EthTxs)-1]
+		if lastEthTx.Nonce()+1 > t.accNonce {
+			t.accNonce = lastEthTx.Nonce() + 1
 		}
 		if receipt.Status == types.ReceiptStatusFailed {
 			batchInfo.Debug.Status = StatusFailed
-			_, err := t.ethClient.EthCall(ctx, batchInfo.EthTx, receipt.BlockNumber)
-			log.Warnw("TxManager receipt status is failed", "tx", receipt.TxHash,
-				"batch", batchInfo.BatchNum, "block", receipt.BlockNumber.Int64(),
-				"err", err)
-			batchInfo.EthTxErr = err
+			var err error
+			for _, tx := range batchInfo.EthTxs {
```

Comment:
Is it okay, that I do ethCall for every tx here?

Reply:
Ok